### PR TITLE
Some housekeeping/small bug fixes

### PR DESCRIPTION
**Pull request by ifndefJOSH**
I'd like to get more into LMMS development by starting with fixing smaller bugs and things like that as I get familiar with the codebase. This PR will contain some of my bug fixes and will be marked as a draft. Currently, there are changes here to address:

- Issue #6173 : in plugins, `enum` is changed to `enum class`
- Issue #6487  : creates a division floor to prevent division by 0
- (Solution not finished yet) Issue #6004 : Currently creates a 16px right margin in the scrollbox. Would rather have the margin be dynamic when the scrollbar appears and I know it's possible in Qt to get scrollbar width. Don't use this solution yet.
    - Also in Qt5 `setMargins()` is depricated, so I used `setContentMargins()`

### CODE REVIEW

**Comment by zonkmachine** [comment]
As for caps, we don't really touch the code in **/plugins/LadspaEffect/** as it's 'upstream'. Caps will eventually be switched to a submodule (https://github.com/LMMS/lmms/pull/4027) so any changes there will be undone at that point.

File: src/gui/editors/TrackContainerView.cpp [hunk 1]
```diff
@@ -89,8 +89,8 @@ TrackContainerView::TrackContainerView( TrackContainer * _tc ) :
 	m_tc->setHook( this );
 	//keeps the direction of the widget, undepended on the locale
 	setLayoutDirection( Qt::LeftToRight );
-	auto layout = new QVBoxLayout(this);
-	layout->setMargin( 0 );
+	QVBoxLayout * layout = new QVBoxLayout( this );
+	layout->setContentsMargins( 0, 0, RIGHT_SCROLLBOX_MARGIN, 0 );
```
Comment (Veratil) [inline src/gui/editors/TrackContainerView.cpp:93]:
```suggestion
	auto layout = new QVBoxLayout(this);
	layout->setContentsMargins(0, 0, RIGHT_SCROLLBOX_MARGIN, 0);
```
This doesn't need to change off `auto`, and adding whitespace is no-no. See [Coding guidelines](https://github.com/LMMS/lmms/wiki/Coding-Conventions#whitespace).

File: include/BandLimitedWave.h [hunk 2]
```diff
@@ -101,6 +101,7 @@ class LMMS_EXPORT BandLimitedWave
 	BandLimitedWave() = default;
 	virtual ~BandLimitedWave() = default;
 
+	static const float DIVISION_FLOOR = 1e-4; // This can be changed
```
Comment (Veratil) [inline include/BandLimitedWave.h:104]: We'll need to figure out a good small number here. I'm not sure if 0.0001 is suitable.

**Review by Veratil — CHANGES_REQUESTED** [review]

File: src/gui/editors/TrackContainerView.cpp (hunk 1 above)
Comment (ifndefJOSH) [reply src/gui/editors/TrackContainerView.cpp]: Sorry, the `auto` change must have been when I merged my commit with some upstream commits I hadn't pulled down

File: include/BandLimitedWave.h (hunk 2 above)
Comment (ifndefJOSH) [reply include/BandLimitedWave.h:104]: If we're dividing by a frequency, we could make it as high as 1

**Comment by allejok96** [comment]
By changing to `enum class` you must make sure to write the full type everywhere the enum is used.

As you can see the build fails now. For example in `Monstro.h` you must change this:

```diff
- return BandLimitedWave::oscillate( _ph, _wavelen, BandLimitedWave::BLTriangle );
+ return BandLimitedWave::oscillate(_ph, _wavelen, BandLimitedWave::Waveform::BLTriangle);
```

There are some cases where enum class becomes really important (like when there is a class with the same name as in the enum), and it is generally a good idea to not have a ton of names in the same scope to avoid collision.

But on the other hand enum class makes code longer and switching to it introduces changes everywhere so unless there's a good reason for a particular enum to change, or we agree to change *all* enums in one go, I don't think we need to touch those right now. (That's just a personal opinion)

**Comment by luzpaz** [comment]
bump @ifndefJOSH

**Comment by DomClark** [comment]
The enum class changes I have now done in #6760.

**Pull request closed without merging**